pallets/proxy-keystore/src/lib.rs

```diff
@@ -0,0 +1,311 @@
+// Copyright 2022 Centrifuge Foundation (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use enum_iterator::{all, Sequence};
+use frame_support::pallet_prelude::*;
+pub use pallet::*;
+use scale_info::TypeInfo;
+use sp_std::vec::Vec;
+pub use weights::*;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub mod weights;
+
+// make sure representation is 1 byte
```

Comment:
@mikiquantum hey, can you please provide more details on this?

Reply:
Oh, I am referring to minimize as much as possible the storage needed here to store the enum. I am not sure what is the internal representation of this enum when encoding it and append it to the KeyId